Observer/SourceDeductionProcessor.php

```diff
@@ -87,6 +111,42 @@ public function execute(EventObserver $observer)
 
         $sourceSelectionResult = $this->getSourceSelectionResultFromOrder->execute($order);
 
+        if (!$extensionAttributes = $order->getExtensionAttributes()) {
+            $extensionAttributes = $this->orderExtensionFactory->create();
+        }
+
+        $extensionAttributes->setSources(
+            $sourcesItems = $sourceSelectionResult->getSourceSelectionItems()
+        );
+
+        $order->setExtensionAttributes($extensionAttributes);
+
+        $sources = [];
+        foreach ($sourcesItems as $item) {
+            $sources[] = [
+                'source_code' => $item->getSourceCode(),
+                'SKU' => $item->getSku(),
+                'qty_to_deduct' => $item->getQtyToDeduct(),
+                'qty_available' => $item->getQtyAvailable()
+            ];
+        }
+
+        $model = $this->sourcesFactory->create();
+        $this->sourcesResourceModel->load($model, $orderId = $order->getId(), 'order_id');
+
+        if (!$model->getId()) {
+            $model->addData(
+                [
+                    'order_id' => $orderId,
+                    'sources' => $this->serializer->serialize($sources)
+                ]
+            );
+        } else {
+            $model->setData('sources', $this->serializer->serialize($sources));
+        }
+
+        $this->sourcesResourceModel->save($model);
+
```

Comment:
So there's now no mechanism that loads the data? We still need to load the sources data with the order, likely through repository methods and the such

Reply:
Yo, so this still seems like an issue to me, we're not actually loading the extension attributes with an order? We're simply saving them. Don't we need to set this data against an order when it's loaded?